clang/runtime/dpct-rt/include/dpl_extras/vector.h

```diff
@@ -325,57 +418,62 @@ class device_vector {
   void insert(iterator position, size_type n, const T &x) {
     if (position == end()) {
       resize(size() + n);
-      ::std::fill(oneapi::dpl::execution::make_device_policy(get_default_queue()),
-                end() - n, end(), x);
+      ::std::fill(
+          oneapi::dpl::execution::make_device_policy(get_default_queue()),
+          end() - n, end(), x);
     } else {
       auto i_n = ::std::distance(begin(), position);
       // allocate temporary storage
       auto m = ::std::distance(position, end());
       // will throw if position is not inside active vector
-      auto tmp = _alloc.allocate(m);
+      auto tmp = ::std::allocator_traits<Allocator>::allocate(_alloc, m);
```

Comment:
Why it will throw?

Reply:
It will throw because it would be attempting to allocate a negative number of bits for the temporary.

Reply:
Is it possible that the the distance is a positive number and not to large, so the allocation can be done?

Reply:
Yes, in normal operation where this position *is* inside the vector, it the distance will return a reasonable positive number and the allocation will succeed.